tree-sitter-stack-graphs/src/lib.rs

```diff
@@ -432,49 +434,129 @@ impl StackGraphLanguage {
     /// nodes and edges in `stack_graph`.  Any new nodes that we create will belong to `file`.
     /// (The source file must be implemented in this language, otherwise you'll probably get a
     /// parse error.)
-    pub fn build_stack_graph_into(
-        &self,
-        stack_graph: &mut StackGraph,
+    pub fn build_stack_graph_into<'a>(
+        &'a self,
+        stack_graph: &'a mut StackGraph,
+        file: Handle<File>,
+        source: &'a str,
+        globals: Variables<'a>,
```

Comment:
This used to be `&mut Variables`, mostly for technical reasons. But it made the API a bit unclear in what could happen to the value. Was it intended to be mutated, was it guaranteed not to be? The current code _would_ mutate in such a way that reusing it for another call to `build_stack_graph_into` resulted in a panic. By passing in the value, its clear to the caller what is going to happen, and it gives the library a bit more flexibility (e.g., by being able to reuse the value instead of creating a nested context just in case).